src/NeoSharp.Core/DI/Modules/NetworkModule.cs

```diff
@@ -28,6 +29,8 @@ public void Register(IContainerBuilder containerBuilder)
             containerBuilder.RegisterSingleton<IPeerListener, TcpPeerListener>();
             containerBuilder.RegisterSingleton<ITcpPeerFactory, TcpPeerFactory>();
 
+            containerBuilder.RegisterInstanceCreator<IEnumerable<IBroadcast>>(c => new IBroadcast[] { c.Resolve<IServer>() });
```

Comment:
It always contains a single item which is IServer, right?

Reply:
My couple thoughts were why to define a collection from a single element and should be broadcast incorporated inside of Server class to propagate messages over connectedPeers collection